client/src/features/forgotpassword/ForgotPasswordReset.tsx

```diff
@@ -1,101 +1,177 @@
-			<LeftPart />
-			<section className='flex justify-center my-auto h-9/12 w-1/2 tablet_l_max:w-full '>
-				<div className='flex flex-col justify-end h-full w-7/12 mobile_xl_max:w-9/12'>
-					<p className='flex-initial mb-8 mobile_m_max:text-2xl mobile_m:text-3xl tablet_s:text-4xl'>
-						Create new password
-					</p>
-					<form action='' className='flex flex-col w-full'>
-						<label htmlFor='password' className='text-lg mb-2'>
-							Password
-						</label>
-						<div className='relative'>
-							<input
-								type={isVisible ? 'text' : 'password'}
-								className='h-10 mb-12 w-full px-8'
-								onChange={(e) =>
-									setPasswords({ ...passwords, password: e.target.value })
-								}
-								value={passwords.password}
-							/>
-							<FaUnlockAlt className='absolute top-2.5 left-2 text-xl text-gray-300' />
-							<AiFillEyeInvisible
-								className={[
-									'cursor-pointer',
-									'absolute',
-									'top-2',
-									'left-custom_left',
-									'mobile_l_max:left-mb_custom_left',
-									'text-2xl',
-									'text-gray-300',
-									isVisible ? 'hidden' : null,
-								].join(' ')}
-								onClick={() => setIsVisible((prevState) => !prevState)}
-							/>
-							<AiFillEye
-								className={[
-									'cursor-pointer',
-									'absolute',
-									'top-2',
-									'left-custom_left',
-									'mobile_l_max:left-mb_custom_left',
-									'text-2xl',
-									'text-gray-300',
-									isVisible ? null : 'hidden',
-								].join(' ')}
-								onClick={() => setIsVisible((prevState) => !prevState)}
-							/>
-						</div>
-						<div className='relative'>
-							<input
-								type='password'
-								className='h-10 mb-12 w-full px-8'
-								onChange={(e) =>
-									setPasswords({
-										...passwords,
-										confirmedPassword: e.target.value,
-									})
-								}
-								value={passwords.confirmedPassword}
-							/>
-							<FaUnlockAlt className='absolute top-2.5 left-2 text-xl text-gray-300' />
-						</div>
-						<button
-							className='button mb-10'
-							onClick={(e) => createNewPassword(e)}
-						>
-							Create
-						</button>
-						<Link to='/login' className='flex-initial underline m-auto'>
-							Back to login
-						</Link>
-					</form>
-				</div>
-			</section>
-		</main>
-	);
+  useEffect(() => {
+    const verifyUrl = async () => {
+      try {
+        console.log('verifying');
+        const response = await axios.get(
+          `http://localhost:5000/forgotpassword/reset/${hasheduserid}`
+        );
+        console.log(response.data.message);
+        setIsUrlValid(true);
+      } catch (error: any) {
+        console.log(`the url is already expired! ${error}`);
+        if (error.response) {
+          console.error(error.response.data.message);
+        } else {
+          console.error(error);
+        }
+        setIsUrlValid(false);
+      }
+    };
```

Comment:
**Suggestion:** place the function `verifyUrl` outside of `useEffect` or invoke this function immediately 💡